DQM/BeamMonitor/plugins/BeamSpotDipServer.cc

```diff
@@ -0,0 +1,770 @@
+void BeamSpotDipServer::problem()
+{
+  if(verbose)
+    edm::LogInfo("BeamSpotDipServer") << "no update | alive = " << alive;
+
+  lsCount++;
+
+  if ((lsCount % timeoutLS[0] == 0)
+   && (lsCount % timeoutLS[1] != 0)) // first time out
+  {
+    if (!alive.test(1)) alive.flip(1);
+    if (!alive.test(2))
+    {
+      if (!alive.test(7)) fakeRcd();
+                     else trueRcd();
+
+      stringstream warnMsg;
+      warnMsg << "No new data for " << lsCount << " LS";
+      publishRcd("Uncertain",warnMsg.str(),false,false);
+    } else {
+      fakeRcd();
+
+      stringstream warnMsg;
+      warnMsg << "No new data for " << lsCount << " LS: "
+               << tkStatus();
+      publishRcd("Bad",warnMsg.str(),false,false);
+    }
+  }
+  else if (lsCount % timeoutLS[1] == 0) // second time out
+  {
+    if (!alive.test(2)) alive.flip(2);
+    fakeRcd();
+
+    stringstream warnMsg;
+    warnMsg << "No new data for " << lsCount << " LS: " << tkStatus();
+    publishRcd("Bad",warnMsg.str(),false,false);
+  }
+}
+
+/*****************************************************************************/ 
+bool BeamSpotDipServer::readRcd(const BeamSpotOnlineObjects & bs)
+// read from database
+{
+  runnum = bs.GetLastAnalyzedRun();
+
+  // BeamSpotOnlineObject only provides creation time
+
+  startTime      = bs.GetCreationTime(); // provides cond::Time_t -> string ?
+  startTimeStamp = bs.GetCreationTime(); // provides cond::Time_t -> time_t ?
+    endTime      = bs.GetCreationTime();
+    endTimeStamp = bs.GetCreationTime();
+
+  lumiRange = to_string(bs.GetLastAnalyzedLumi());
+  currentLS = bs.GetLastAnalyzedLumi();
+
+  type = bs.GetBeamType();
+
+  if(verbose)
+    edm::LogInfo("BeamSpotDipServer")
+         << "run: "    << runnum
+         << ", LS: "   << currentLS 
+         << ", time: " << startTime << " " << startTimeStamp
+         << ", type: " << type;
+
+  if (testing)          quality = qualities[0]; // Uncertain
+  else if (type >= 2)   quality = qualities[2]; // Good
+                   else quality = qualities[1]; // Bad
+
+  x = bs.GetX();
+  y = bs.GetY();
+  z = bs.GetZ();
+
+  sigma_z = bs.GetSigmaZ();
+  dxdz    = bs.Getdxdz(); 
+  dydz    = bs.Getdydz();
+  width_x = bs.GetBeamWidthX();
+  width_y = bs.GetBeamWidthX();
+
+  err_x       = bs.GetXError();
+  err_y       = bs.GetYError();
+  err_z       = bs.GetZError();
+  err_sigma_z = bs.GetSigmaZError();
+  err_dxdz    = bs.GetdxdzError();
+  err_dydz    = bs.GetdydzError();
+  err_width_x = bs.GetBeamWidthXError();
+  err_width_y = bs.GetBeamWidthYError();
+
+      events = 0.; // not provided by BeamSpotOnlineObject
+      meanPV = 0.; // not provided by BeamSpotOnlineObject
+  err_meanPV = 0.; // not provided by BeamSpotOnlineObject
+       rmsPV = 0.; // not provided by BeamSpotOnlineObject
+   err_rmsPV = 0.; // not provided by BeamSpotOnlineObject
+       maxPV = 0.; // not provided by BeamSpotOnlineObject
```

Comment:
Methods are now provided by the [BeamSpotOnlineObjects class](https://github.com/cms-sw/cmssw/blob/master/CondFormats/BeamSpotObjects/interface/BeamSpotOnlineObjects.h):
- GetUsedEvents()
- GetMeanPV()
- GetMeanErrorPV()
- GetRmsPV()
- GetRmsErrorPV()
- GetMaxPVs()